wpical/src/main/native/include/fieldcalibration.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <ceres/ceres.h>
+#include <opencv2/calib3d.hpp>
+#include <opencv2/core/eigen.hpp>
+#include <opencv2/core/utils/logger.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
+#include <wpi/json.h>
+
+#include "apriltag.h"
+
+struct Pose {
+  Eigen::Vector3d p;
+  Eigen::Quaterniond q;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+struct Constraint {
+  int id_begin;
+  int id_end;
+  Pose t_begin_end;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+namespace fieldcalibration {
+std::tuple<Eigen::Matrix<double, 3, 3>, Eigen::Matrix<double, 8, 1>>
+load_camera_model(std::string path);
+std::tuple<Eigen::Matrix<double, 3, 3>, Eigen::Matrix<double, 8, 1>>
+load_camera_model(wpi::json json_data);
+std::map<int, wpi::json> load_ideal_map(std::string path);
+std::vector<Eigen::Vector3d> get_model_corners(double tag_size);
+Eigen::Matrix<double, 4, 4> get_tag_transform(
+    std::map<int, wpi::json>& ideal_map, int tag_id);
+Eigen::Matrix<double, 4, 4> estimate_tag_pose(
+    apriltag_detection_t* tag_detection,
+    const Eigen::Matrix<double, 3, 3>& camera_matrix,
+    const Eigen::Matrix<double, 8, 1>& camera_distortion, double tag_size);
+void draw_tag_cube(cv::Mat& frame, Eigen::Matrix<double, 4, 4> camera_to_tag,
+                   const Eigen::Matrix<double, 3, 3>& camera_matrix,
+                   const Eigen::Matrix<double, 8, 1>& camera_distortion,
+                   double tag_size);
+bool process_video_file(
```

Comment:
Is process_video_file intended to be part of your external API surface? I only see it being used in fieldcalibration.cpp. If not, consider making static.

Reply:
All of the functions except calibrate are only ever used in fieldcalibration. Should they all be implemented statically in fieldcalibration.h?

Reply:
If a function doesn't need to be accessible outside of a single .cpp file, they should be declared as static functions in that cpp file.